### PR TITLE
[MAINTENANCE] Helpful Message for Expectations that require Metrics that are not available

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
Currently, if an Expectation is misconfigured (such as referencing a non-existent column name), we get a cryptic error message.

```python 
    provided_metrics: Dict[str, MetricValue] = {
>       metric_name: metrics[metric_configuration.id]
        for metric_name, metric_configuration in validation_dependencies.metric_configurations.items()
    }
E   KeyError: ('column_values.nonnull.unexpected_count', '9fb963653447edb4ae8538917d6915bc', ())
```

This PR adds a check in `expectation.py`, where we validate dependencies of the current Expectation against available metrics, and outputs a more helpful error message: 

An error message will look something like this: 
```python
great_expectations.exceptions.exceptions.InvalidExpectationConfigurationError: Metric ('table.row_count', '7054e9d72e974fad769b0964f41e7e5b', "result_format={'result_format': 'BASIC', 'partial_unexpected_count': 20, 'include_unexpected_rows': False}") is not available for validation of {
                 "expectation_type": "expect_column_values_to_not_be_null",
                 "kwargs": {
                   "column": "I_dont_exist",
                   "batch_id": "projects-projects"
                 },
                 "meta": {}
              }. Please check your configuration.
```

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

### CODE REVIEW

File: tests/expectations/test_expectation.py [hunk 1]
```diff
@@ -20,10 +21,42 @@ class FakeColumnPairMapExpectation(expectation.ColumnPairMapExpectation):
...
+    return [
+        MetricConfiguration(
+            metric_name=metric_name,
+            metric_domain_kwargs=metric_domain_kwargs,
+            metric_value_kwargs={},
+        )
+    ]
+
+
+def fake_expectation_config(
```
Comment (anthonyburdi) [inline tests/expectations/test_expectation.py:53]: Good names make things so much easier :heart: Thank you for refactoring this.

**Comment by JennyTee** [comment]
I like the extra error message; that seems much more helpful. Is this the only fix needed for https://github.com/great-expectations/great_expectations/issues/8619, then? Does this mean that for databases with case-insensitive column names, GX users will need to define expectations with case-sensitive column names?

**Review by anthonyburdi — COMMENTED** [review]
Looks great - just wondering about the `pythonsqlite.db-journal` file added. Could you clarify why that was added?

**Review by anthonyburdi — APPROVED** [review]
Very helpful error message, thanks for this PR!

**Pull request merged**